classes/components/listPanels/PKPSubmissionsListPanel.inc.php

```diff
@@ -261,4 +263,65 @@ protected function _getItemsParams() {
 			$this->getParams
 		);
 	}
+
+	/**
+	 * Compile the categories for passing as filters
+	 *
+	 * @return array
+	 */
+	public function getCategoryFilters() {
+		$request = \Application::get()->getRequest();
+		$context = $request->getContext();
+
+		$categories = [];
+		$categoryDao = \DAORegistry::getDAO('CategoryDAO')
+		; /* @var $categoryDao CategoryDAO */
+		$categoryIterator = $categoryDao->getByContextId($context->getId())->toAssociativeArray();
+		foreach ($categoryIterator as $category) {
+			$categories[] = array(
+				'id' => $category->getId(),
+				'title' => $category->getLocalizedTitle(),
+			);
+		}
```

Comment:
Because there are four submission lists on the dashboard (my queue, unassigned, all active, archived), this code will be run four times. On an install with a lot of categories, that could get pretty slow.

Can you try passing in the categories to each ListPanel so that you only have to get the categories once? You can see how this is done with the `includeAssignedEditorsFilter` here:

https://github.com/pkp/pkp-lib/blob/main/pages/dashboard/DashboardHandler.inc.php#L119

The data is stored in the list panel when a property is defined for it:

https://github.com/pkp/pkp-lib/blob/main/classes/components/listPanels/PKPSubmissionsListPanel.inc.php#L41